src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java

```diff
@@ -51,13 +52,30 @@ public Optional<Path> copyFileToFileDir(Path file) {
         Optional<Path> firstExistingFileDir = bibDatabaseContext.getFirstExistingFileDir(filePreferences);
         if (firstExistingFileDir.isPresent()) {
             Path targetFile = firstExistingFileDir.get().resolve(file.getFileName());
+            if (file.equals(targetFile)) {
```

Comment:
Equals on Path ca be misleading we probably want ot check the filename only

Reply:
59f38fb10a85e8a5e1a0519c2fe9f2539586ba78